From 92d7a2fea128bd727111f05d39311dd43c18ada4 Mon Sep 17 00:00:00 2001
From: Moritz Vogel <moritzv7@gmail.com>
Date: Thu, 25 Apr 2024 15:13:14 +0200
Subject: [PATCH] Prevent Steam from inhibiting the screensaver

---
 main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/main.c b/main.c
index 6511d3e..7b0c506 100644
--- a/main.c
+++ b/main.c
@@ -560,6 +560,11 @@ static void setup_property_changed_listener(void) {
 }
 
 static uint32_t handle_screensaver_inhibit(const char *sender, const char *app_name, const char *reason) {
+	if (strcmp(app_name, "Steam") == 0 || strcmp(app_name, "My SDL Application") == 0) {
+		swayidle_log(LOG_INFO, "Ignoring " DBUS_SCREENSAVER_INTERFACE ".Inhibit('%s', '%s') from %s", app_name, reason, sender);
+		return 0;
+	}
+
 	swayidle_log(LOG_INFO, DBUS_SCREENSAVER_INTERFACE ".Inhibit('%s', '%s') from %s", app_name, reason, sender);
 
 	uint32_t cookie;
-- 
2.44.0

