From ab009e0cec61c770c3e2f7ea193200f8a24954f3 Mon Sep 17 00:00:00 2001
From: Moritz Vogel <moritzv7@gmail.com>
Date: Sun, 21 Jan 2024 00:07:41 +0100
Subject: [PATCH 1/2] Show dirsize if dircounts is enabled

---
 ui.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ui.go b/ui.go
index 4c85aec..80b1088 100644
--- a/ui.go
+++ b/ui.go
@@ -306,6 +306,11 @@ func fileInfo(f *file, d *dir, userWidth int, groupWidth int) string {
 				default:
 					info.WriteString(" 999+")
 				}
+
+				if f.dirSize > 0 {
+					sz := humanize(f.TotalSize())
+					info = fmt.Sprintf("[%4s]%s", sz, info)
+				}
 				continue
 			}
 
-- 
2.47.0

