From 0fd004803e9347fb8760c34b7b939d8c247bea6b Mon Sep 17 00:00:00 2001
From: Moritz Vogel <moritzv7@gmail.com>
Date: Sun, 21 Jan 2024 00:07:41 +0100
Subject: [PATCH 1/2] Show dirsize if dircounts is enabled

---
 ui.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ui.go b/ui.go
index c35b6f9..3fcf90d 100644
--- a/ui.go
+++ b/ui.go
@@ -301,6 +301,11 @@ func fileInfo(f *file, d *dir, userWidth int, groupWidth int, customWidth int) (
 		switch s {
 		case "size":
 			if f.IsDir() && getDirCounts(d.path) {
+				if f.dirSize > 0 {
+					sz := humanize(uint64(f.TotalSize()))
+					fmt.Fprintf(&info, "[%4s] ", sz)
+				}
+
 				switch {
 				case f.dirCount < -1:
 					info.WriteString("     !")
-- 
2.50.1

